tests/framework/extensions/clusters/clusters.go

```diff
@@ -1,11 +1,35 @@
 package clusters
 
 import (
+	"context"
+	"fmt"
+	"strings"
+
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
+func ClusterReadyStatusFunc(event watch.Event) (ready bool, err error) {
+	cluster := event.Object.(*apisV1.Cluster)
+	count := 0
+
+	for _, condition := range cluster.Status.Conditions {
+		if strings.Contains(condition.Reason, "Error") {
+			count += 1
+			if count == 10 {
```

Comment:
It's an arbitrary number, I don't remember why I put 10. We can change it

Reply:
So I am iterating over all of them. This counter is an error threshold, as there are times the cluster may have an error, but it will resolve itself. This is to prevent a false error.